src/VersionUpdater/Installer.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+/*
+ * This file is part of Psy Shell.
+ *
+ * (c) 2012-2022 Justin Hileman
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Psy\VersionUpdater;
+
+use Psy\Exception\ErrorException;
+
+class Installer
+{
+    /**
+     * @var string
+     */
+    protected $installLocation;
+
+    /**
+     * @var string
+     */
+    protected $tempDirectory;
+
+    public function __construct(string $tempDirectory = null)
+    {
+        $this->tempDirectory = $tempDirectory ?: \sys_get_temp_dir();
+        $this->installLocation = \realpath($_SERVER['argv'][0]);
```

Comment:
should we use `\Phar::running(true)`?

Reply:
For sure, but it would have to be `\Phar::running(false)` since fs operations won't work with 'phar://' uri's

Reply:
sorry, yeah. got the argument flipped.